panoptes_cli/commands/subject_set.py

```diff
@@ -414,6 +507,23 @@ def link_subjects(limit):
                     subject = Subject()
                     subject.links.project = subject_set.links.project
                     for media_file in files:
+                        file_size = os.path.getsize(media_file)
+                        if (file_size > MAX_UPLOAD_FILE_SIZE) and \
+                                (upload_state['compress']):
+                            # compress with lower quality until the size fits
+                            input_file = media_file
+                            for n in range(11):
+                                quality = 80 - n * 3
+                                media_file = compress_image(input_file,
+                                                            quality)
+                                new_file_size = media_file.getbuffer().nbytes
+                                if new_file_size <= MAX_UPLOAD_FILE_SIZE:
+                                    break
+                            if new_file_size > MAX_UPLOAD_FILE_SIZE:
+                                raise ValueError(
```

Comment:
This should use `click.echo` to print an error message and then return -1, rather than raising an exception.